# BIOME METAGENOMIC PUBLIC DATABASE

## Introduction  
  
The BIOME metagenomic public database contains 5625 metagenomic samples collected from MG-RAST and NCBI databases. The samples were recategorized based on two methods, the authoral BIOME classification and the IUCN classification.  
These datbase comprimises other works of the lab, such as:
- [Aquifers Landscapes](https://github.com/MeirellesLab/aquifer_metagenomes)
- [New Microbial Keystones](https://github.com/MeirellesLab/keystones_paper)

**warning: None of the present data should be shared or used without the consent of the supervisor Dr. Pedro Meirelles.**

## Data organization
- In the [annotated_metagenomes](annotated_metagenomes/) you will find the taxonomic relative abundance matrices generated by Kraken2 using the Biome database.  
- in the [metadata](metadata/) you will find the raw and treated versions of the MG-RAST, BIOME and IUCN classifications.  
- in the [genomic_content_summaries](genomic_content_summaries/) folder you will find the genomic content information of our samples.  
- in the [treating_data](treating_data/) folder you will find the pipelines we used to produce our final metadata. (control porpuses only)  

# 1. Colection and Classification Protocol 
This step resolves the metadata and final samples to be used in the database.

## 1.1 Coleção  
- Coletamos as amostras no mg-rast de shotgun metagenomes [COLOCAR DATA] (n= 70,000)  
- Essas amostras foram categorizadas subjetiva e grosseiramente em estilo de vida, ecosistema e habitats;  
- O resultado foi a tabela [mg-rast_metadata](reclassification_2022/01_original_data/mg-rast_metadata.csv) (n= 7,044);  
- No mesmo período coletamos amostras de aquíferos da plataforma NCBI e focam curadas manualmente (n = 109)  
- Resultado: [aquifer_samples](reclassification_2022/01_original_data/aquifer_samples.csv) (n= 109);  

    
## 1.2 Curadoria baseada no metadado do Mg-Rast
- A partir do [mg-rast_metadata](reclassification_2022/01_original_data/mg-rast_metadata.csv), baixamos o json do metadado de cada metagenoma para a reclassificação fina das amostras utilizando o [script](Python/mgrast_download_metadata.py) e unimos as tabelas removendo as amostras com metagenomas assembled a partir do [script](R/merge_tables.R);  
- O resultado foi a tabela [merged_metadata_raw](reclassification_2022/04_final_merge_and_reclassification/biome_classification/merged_metadata_raw.csv)  

## 1.3 Filtragem baseado nos sumários de dados genômicos
- Após baixadas todas as amostras, empregamos filtros de qualidade genômicos, (mais detalhes [aqui](link_do_download) ) e sumarizamos as informações genômicas de cada amostra em uma [tabela](genomic_content_summaries/genomic_content_summaries.csv) (n= 6,015). As informações sumarizadas foram:  
```
file_size                       total_number_of_reads
GC_content                      total_number_of_bases
complete_sequence_size          average_read_size
read_size_mode                  read_size_median
lowest_read_size                highest_read_size
```
- A partir deste dado sumarizado, removemos todas as amostras a partir de 10 mil reads

# 2. Download of the samples

























































  
### 1. Biome classification information  

#### The BIOME classification is based on the ecologic and geographic information of the samples. There we have:  
  
##### 2 life-styles   
```
free-living         holobiont
  
```
##### 3 biospheres  
```
freshawater         marine                  terrestrial
```
  
##### 9 ecosystems  
```
wastewater          animal_host-associated  saline_water 
soil                freshwater              human_host-associated
plant-associated    sediment                groundwater
```

##### 85 habitats  
                                                                                                                                                       
```
coral           karst_porous              plant-associated_hypersaline_water
river           shrubland_soil            continental_slope_sediment
fish            bone                      marine_biofilm
sponge          saline_marsh              saline_marsh_sediment
compost         polar_seawater            oil_affected_sediment
rhirosphere     animal_skin               human_associated_biofilm
porous          geyser                    saline_evaporation_pond
sea_worm        human_feces               subsurface_saline
chyme           hypersaline_water         hydrothermal_vent
tundra_soil     street_sweepings          aqueous_humour
farm_soil       anoxic_seawater           coral_reef_biofilm
rumen           wood_decomposition        coastal_marine_sediment
salt_lake       oceanic_seawater          estuarine_seawater
human-skin      contaminated_soil         coastal_seawater
human-gut       bodily_fluid              mangrove_sediment
animal_feces    river_saline              agricultural_soil
meadow_soil     savanna_soil              peat_soil
park_soil       contaminated_water        forest_soil
animal_manure   desert_sediment           pasture_soil
desert_soil     sulfur_spring             wastewater_treatment_plant
lung            freshwater_biofilm        alkaline_environment
animal_gut      hot_spring                drinking_water_reservoir
garden_soil     stream_biofilm            saline_water_biofilm
mine            permafrost                plant-associated
mummy           lake_sediment             coral_reef_seawater
saliva          reactor_sludge            anaerobic_sediment
grassland       river_sediment            sewage
lake            stream                    porous_contaminated
sludge

```







